pennylane/compiler/qjit_api.py

```diff
@@ -111,3 +111,173 @@ def circuit(theta):
     compilers = AvailableCompilers.names_entrypoints
     qjit_loader = compilers[compiler]["qjit"].load()
     return qjit_loader(fn=fn, *args, **kwargs)
+
+
+def while_loop(cond_fn):
+    """A :func:`~.qjit` compatible while-loop for PennyLane programs.
+
+    .. note::
+
+        This function only supports the Catalyst compiler. See
+        :func:`catalyst.while_loop` for more details.
+
+        Please see the Catalyst :doc:`quickstart guide <catalyst:dev/quick_start>`,
+        as well as the :doc:`sharp bits and debugging tips <catalyst:dev/sharp_bits>`
+        page for an overview of the differences between Catalyst and PennyLane.
+
+    This decorator provides a functional version of the traditional while
+    loop, similar to ``jax.lax.while_loop``. That is, any variables that are
```

Comment:
(optional) Could link out to JAX here

Reply:
Thanks @trbromley! Fixed [784e532](https://github.com/PennyLaneAI/pennylane/pull/4698/commits/784e53253ff3fc306019642a71007e79e5dd6b98)